modelarmor/create_template_with_metadata.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample code for creating a new model armor template with template metadata.
+
+package modelarmor
+
+// [START modelarmor_create_template_with_metadata]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	modelarmor "cloud.google.com/go/modelarmor/apiv1"
+	modelarmorpb "cloud.google.com/go/modelarmor/apiv1/modelarmorpb"
+	"google.golang.org/api/option"
+)
+
+// createModelArmorTemplateWithMetadata method creates a
+// new Model Armor template with template metadata.
+//
+// w io.Writer: The writer to use for logging.
+// projectID string: The ID of the Google Cloud project.
+// locationID string: The ID of the Google Cloud location.
+// templateID string: The ID of the template to create.
+// metadata *modelarmorpb.TemplateMetadata: The template metadata to apply.
+func createModelArmorTemplateWithMetadata(w io.Writer, projectID, locationID, templateID string) error {
+	ctx := context.Background()
+
+	// Create options for Model Armor client.
+	opts := fmt.Sprintf("modelarmor.%s.rep.googleapis.com:443", locationID)
+	// Create the Model Armor client.
+	client, err := modelarmor.NewClient(ctx,
```

Comment:
See previous comment about calling `option.WithEndpoint()` outside of the new client instantiation.